cub/test/catch2_test_triple_chevron.cu

```diff
@@ -0,0 +1,211 @@
+                                               bool device_invoke)
+  {
+    return invoke(d_temp_storage,
+                  temp_storage_bytes,
+                  mult_two_kernel<T>,
+                  d_in,
+                  d_out,
+                  num_items,
+                  device_invoke);
+  }
+};
+
+struct cdp_invocable { 
+
+  template <class T>
+  CUB_RUNTIME_FUNCTION cudaError_t operator()(uint8_t* temp, size_t bytes, T d_in, T d_out, int n, bool on_device) const {
+       return cdp_chevron_invoker::create(
+          temp,
+          bytes, 
+          d_in, 
+          d_out,
+          n, 
+          on_device
+       );
+  }
+};
+
+
+__global__ void add_kernel(int a, float b, double* out) {
+    *out = a + b;
+}
+
+CUB_TEST("CDP wrapper works with custom invocables and cdp_launch, on both host and device", "[test][utils]")
+{
+  int n = 42;
+  thrust::device_vector<int> in(n, 21);
+  thrust::device_vector<int> out(n);
+
+  int *d_in  = thrust::raw_pointer_cast(in.data());
+  int *d_out = thrust::raw_pointer_cast(out.data());
+
+  constexpr bool on_device = TEST_CDP;
+
+  {
+    cdp_launch(cdp_invocable{}, d_in, d_out, n, on_device);
+    cudaError_t err = cudaGetLastError();
+    if (err != cudaSuccess) {
+        printf("Error: %s\n", cudaGetErrorString(err));
+    }
+    cudaDeviceSynchronize();
+    std::vector<int> h_out(n);
+    thrust::copy(out.begin(), out.end(), h_out.begin());
+    for(int i = 0; i < std::min(5, n); ++i) {
+        printf("h_out[%d] = %d\n", i, h_out[i]);
+    }
+    const auto actual   = static_cast<std::size_t>(thrust::count(out.begin(), out.end(), 42));
+    printf("Thrust count result: %zu\n", actual);
+    const auto expected = static_cast<std::size_t>(n);
+
+    REQUIRE(actual == expected);
+  }
+
+
+}
+
+
+CUB_TEST("Rough draft of testing Chevron launches successfully ", "[test][utils]") {
+  cudaStream_t stream1, stream2;
+  cudaStreamCreate(&stream1);
+  cudaStreamCreate(&stream2);
+  
+  cudaEvent_t event;
+  cudaEventCreate(&event);
+
+  int n = 1000;
+  thrust::device_vector<int> in(n, 5);
+  thrust::device_vector<int> out(n);
+
+  int *d_in  = thrust::raw_pointer_cast(in.data());
+  int *d_out = thrust::raw_pointer_cast(out.data());
+
+  cudaMemcpyAsync(d_out, d_in, n * sizeof(float), cudaMemcpyHostToDevice, stream1); 
+  cudaEventRecord(event, stream1); 
+
+  const int block_size = 256;
+  const int grid_size = (n * block_size - 1) / block_size;
+  auto chev = cub::detail::triple_chevron(grid_size, block_size, 0, stream2);
+
+  cudaStreamWaitEvent(stream2, event, 0); // sync streams
+  chev.doit(mult_two_kernel<int>, d_in, d_out, n);
+
+  {
+    const auto actual   = static_cast<std::size_t>(thrust::count(out.begin(), out.end(), 10));
+    const auto expected   = static_cast<std::size_t>(n);
+    REQUIRE(actual==expected);
+  }
+  cudaEventDestroy(event);
+
+}
+
+CUB_TEST("Triple Chevron with missing configuration returns cudaErrorMissingConfiguration", "[test][utils]") {
```

Comment:
suggestion: the functionality we wanted to test is if the error is returned, so I'd rephrase that:
```suggestion
CUB_TEST("Triple chevron returns kernel launch failures", "[test][utils]") {
```